### ISSUE TITLE
Rankup gives the wrong rank

### ISSUE THREAD

**Issue by dayseemetrollin** [SirFaizdat/Prison#165]
So I set up the ranks via group manager C then B then A then Free, then I set up the ranks through /prisonranks (rank) (price) (colorname) when I was testing it I started off in C like I was supposed to then I did /rankup and got to b but when I tried ranking up to A it would send me back to C and when I forcibly added myself to B it ranked up to A like its supposed to, but then when I tried to rankup to Free I had the same problem as I did with A
![2016-07-11_09 26 51](https://cloud.githubusercontent.com/assets/20403494/16738279/e401de7c-4749-11e6-8f66-548e180a1de1.png)

**Comment by dayseemetrollin** [comment]
So In this Image I put myself in C then I ranked up to b and then did /rankup again for a and it didnt show the groupmanager message but instead it just added me to C

**Comment by SirFaizdat** [comment]
Try using the `/pranks set id ` command to put them in proper order, starting at 0.

```
/pranks set id C 0
/pranks set id B 1
/pranks set id A 2
/pranks set id Free 3
```

**Comment by dayseemetrollin** [comment]
It shows "this rank does not exist"

**Comment by SirFaizdat** [comment]
Try using the rank names that you set when you created them.

**Comment by dayseemetrollin** [comment]
Still says this message I named them C B A FREE

**Comment by dayseemetrollin** [comment]
THAT IS SO WEIRD, I changed it instead of /pranks set id C 0 to /prisonranks set id C 0 and that worked

**Comment by dayseemetrollin** [comment]
But it still goes from B to C but says that I have ranked up to A

**Comment by SirFaizdat** [comment]
Try restarting your server (not reloading).

**Comment by dayseemetrollin** [comment]
Its still doing this

**Comment by dayseemetrollin** [comment]
If I gave you the login info and stuff for my multicraft and the server Ip would you mind helping me out?

**Comment by dayseemetrollin** [comment]
Heres the Ip if you want 216.172.108.184:25565

**Comment by SirFaizdat** [comment]
I'm currently looking into the problem, it appears to be a bug with the code.

**Comment by SirFaizdat** [comment]
# Ranks PSA
The ranks system in Prison is fundamentally flawed. The reason why this happens is because the system depends on several other plugins in order to successfully complete the rank-up. This would normally work fine, except for one thing: permissions plugins must be specifically set up to allow ranking up through an arbitrary ladder. However, since they are not built for this kind of thing, they get confused when they're told to only remove a certain rank and add a different rank - even though this seems like a simple task.

There is only one solution to this, and that is to give Prison a built-in and standalone ranks system, without a dependency on permissions plugins. Unfortunately, I do not know how long this will take, but it will be in the Prison v2.3 release.

**Comment by FrontierGamers** [comment]
noice

**Comment by SirFaizdat** [comment]
New rank system confirmed in Prison v2.3. Time to design!

**Comment by Camouflage100** [comment]
As stated many many many times before; we're redesigning the ranks system. Until then we can't really help that the system is flawed. The new system will be released in either the 2.3 or 3 release.
